Classes/DataProcessing/LanguageMenuProcessor.php

```diff
@@ -0,0 +1,333 @@
+     * @param array $conf
+     */
+    protected function stdWrap(&$conf)
+    {
+        if (isset($conf['languageUid.'])) {
+            $conf['languageUid'] = $this->cObj->stdWrap($conf['languageUid'], $conf['languageUid.']);
+        }
+    }
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->menuDefaults['as'] = 'languagemenu';
+        $this->menuDefaults['titleField'] = '';
+    }
+
+    /**
+     * Add loading from sys_languages in case special.value is not set
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function validateConfiguration()
+    {
+        $this->processorConfiguration = array_replace_recursive(
+            [
+                'special' => 'language',
+                'special.' => [
+                    'value.' => [
+                        'ifBlank.' => [
+                            'cObject' => 'COA',
+                            'cObject.' => [
+                                'wrap' => '0|',
+                                '10' => 'CONTENT',
+                                '10.' => [
+                                    'table' => 'sys_language',
+                                    'select.' => [
+                                        'pidInList' => 'root',
+                                        'orderBy' => 'sorting'
+                                    ],
+                                    'renderObj' => 'TEXT',
+                                    'renderObj.' => [
+                                        'wrap' => ',|',
+                                        'field' => 'uid'
+                                    ]
+                                ]
+                            ]
+                        ]
+                    ]
+                ]
+            ],
+            $this->processorConfiguration
+        );
+
+        parent::validateConfiguration();
+    }
+
+    /**
+     * Add more language properties
+     */
+    public function prepareLevelLanguageConfiguration()
+    {
+        parent::prepareLevelLanguageConfiguration();
+
+        if ($this->menuConfig['special'] === 'language') {
+            unset($this->menuLevelConfig['stdWrap.']['cObject.']['10']);
+            unset($this->menuLevelConfig['stdWrap.']['cObject.']['10.']);
+            unset($this->menuLevelConfig['stdWrap.']['cObject.']['20']);
+            unset($this->menuLevelConfig['stdWrap.']['cObject.']['20.']);
+            unset($this->menuLevelConfig['stdWrap.']['cObject.']['70.']['value']);
+            unset($this->menuLevelConfig['stdWrap.']['cObject.']['70.']['listNum.']);
+
+            $this->menuLevelConfig['stdWrap.']['cObject.'] = [
+                '1' => 'LOAD_REGISTER',
+                '1.' => [
+                    'languageUid.' => [
+                        'cObject' => 'TEXT',
+                        'cObject.' => [
+                            'value' => $this->menuConfig['special.']['value'],
```

Comment:
We should introduce a new property on root configuration level for this, in the future we will not likely have a special.value configuration here - maybe a `languages` or `languageIdentifiers` should suit better - we can map this codewise.

Reply:
Changed, now a optional parameter languages is supported